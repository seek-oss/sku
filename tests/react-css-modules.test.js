const path = require('path');
const {
  dirContentsToObject,
  runSkuScriptInDir,
  startAssetServer,
  getAppSnapshot,
} = require('@sku-private/test-utils');

const appDir = path.dirname(
  require.resolve('@sku-fixtures/react-css-modules/sku.config.js'),
);
const distDir = path.resolve(appDir, 'dist');

describe('react-css-modules', () => {
  let closeAssetServer;

  beforeAll(async () => {
    await runSkuScriptInDir('build', appDir);
    closeAssetServer = await startAssetServer(4293, distDir);
  });

  afterAll(() => {
    closeAssetServer();
  });

  it('should create valid app', async () => {
    const app = await getAppSnapshot('http://localhost:4293');
    expect(app).toMatchSnapshot();
  });

  it('should generate the expected files', async () => {
    const files = await dirContentsToObject(distDir);
    expect(files).toMatchSnapshot();
  });

  it('should handle Less and css.js in tests', async () => {
    const { childProcess } = await runSkuScriptInDir('test', appDir);
    expect(childProcess.exitCode).toEqual(0);
  });
});
