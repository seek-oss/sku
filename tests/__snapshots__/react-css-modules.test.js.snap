// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-css-modules should create valid app 1`] = `
SCRIPTS: [
  "/runtime-df7fc58286eea709736e.js",
  "/983-7e081b75ea8ba172af1b.js",
  "/main-8fd671b3b4e8f765f588.js",
]
CSS: [
  "/main-cb8848ae92e1e9d08128.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="stylesheet"
      href="styles[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="eRNyDeb">
        <div
          class="MYunKdE"
          data-automation-text="true"
        >
          Initial render
        </div>
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      []
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: 
           class="MYunKdE"
           data-automation-text="true"
         >
-          Initial render
+          Updated render
         </div>
       </div>
     </div>

`;

exports[`react-css-modules should generate the expected files 1`] = `
{
  "983-7e081b75ea8ba172af1b.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "index.html": SCRIPTS: [
    "/runtime-df7fc58286eea709736e.js",
    "/983-7e081b75ea8ba172af1b.js",
    "/main-8fd671b3b4e8f765f588.js",
  ]
CSS: [
    "/main-cb8848ae92e1e9d08128.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="stylesheet"
      href="styles[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="eRNyDeb">
        <div
          class="MYunKdE"
          data-automation-text="true"
        >
          Initial render
        </div>
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      []
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
  </body>
</html>,
  "main-8fd671b3b4e8f765f588.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main-cb8848ae92e1e9d08128.css": .eRNyDeb {
  display: flex;
}
.eRNyDeb .MYunKdE {
  font-size: 32px;
}
,
  "runtime-df7fc58286eea709736e.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;
